lib/extensions/ar_virtual.rb

```diff
@@ -66,17 +66,36 @@ module ClassMethods
     #
 
     def virtual_delegate(*methods)
-      options = methods.pop
-      unless options.kind_of?(Hash) && options[:to]
+      options = methods.extract_options! || {}
+      unless (to = options[:to])
         raise ArgumentError, 'Delegation needs an association. Supply an options hash with a :to key as the last argument (e.g. delegate :hello, to: :greeter).'
       end
-      delegate(*methods, options)
+
+      if options[:name] && methods.size > 1
+        raise ArgumentError, 'Option :name is only available when defining a single virtual method'
+      end
+
+      location = caller_locations(1, 1).first
+      file, line = location.path, location.lineno
 
       # put method entry per method name.
       # This better supports reloading of the class and changing the definitions
       methods.each do |method|
         method_prefix = virtual_delegate_name_prefix(options[:prefix], options[:to])
-        method_name = "#{method_prefix}#{method}"
+        method_name = options[:name] || "#{method_prefix}#{method}"
+
+        # delegate(method, options)
+        definition = (method =~ /[^\]]=$/) ? 'arg' : '*args, &block'
+        default_value = options[:default] ? "|| #{options[:default]}" : nil
```

Comment:
`default` here is going to be a bit tricky...

If you want it to be a static string, you will have to define it as:

```ruby
virtual_delegate :name, :to => :host, :prefix => true, :default => '"N/A"'
```

Which is fine, because it allows `:default` to be another method, but that is not very obvious to the user/developer this is being done.  Is there an easier way to make this more obvious or user friendly?  Perhaps use two different keys:  `:default_value` and `:default_method` (`:default_value` would do an `.inspect` on the value)?